cql2pgjson/src/main/java/org/folio/cql2pgjson/util/DbSchemaUtils.java

```diff
@@ -54,4 +65,99 @@ public static Index getIndex(String cqlIndex, List<Index> indexes) {
     return null;
   }
 
+  /**
+   * Get {@link Table} for given table name in {@link Schema}
+   *
+   * @param dbSchema
+   * @param tableName
+   * @return
+   */
+  public static Table getTable(Schema dbSchema, String tableName) {
+    for (Table table : dbSchema.getTables()) {
+      if (tableName.equals(table.getTableName())) {
+        return table;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Find a list of {@link ForeignKeys} from source table to target table alias
+   *
+   * @param dbSchema
+   * @param srcTabName
+   * @param targetTabAlias
+   *
+   * @return a list of {@link ForeignKeys}
+   */
+  public static List<DbFkInfo> findForeignKeysFromSourceTableToTargetAlias(Schema dbSchema, String srcTabName,
+      String targetTabAlias) {
+    return findForeignKeys(dbSchema, srcTabName, targetTabAlias, true);
+  }
+
+  private static List<DbFkInfo> findForeignKeys(Schema dbSchema, String srcTabName, String targetTabName,
+      boolean useTargetAlias) {
+    List<DbFkInfo> list = new ArrayList<>();
+    Table srcTab = getTable(dbSchema, srcTabName);
+    if (srcTab.getForeignKeys() == null) {
+      return list;
+    }
+    // direct FK
+    for (ForeignKeys fk : srcTab.getForeignKeys()) {
+      String targetName = useTargetAlias ? fk.getTargetTableAlias() : fk.getTargetTable();
+      if (targetTabName.equals(targetName)) {
+        list.add(new DbFkInfo(srcTab.getTableName(), fk.getFieldName(), fk.getTargetTable()));
+        return list;
+      }
+    }
+    // find the shortest path
+    for (ForeignKeys fk : srcTab.getForeignKeys()) {
+      updateFkList(list, dbSchema, srcTab, fk, targetTabName, useTargetAlias);
+    }
+    return list;
+  }
+
+  private static void updateFkList(List<DbFkInfo> list, Schema dbSchema, Table srcTab, ForeignKeys fk,
+      String targetTabName, boolean useTargetAlias) {
+    List<DbFkInfo> childList = findForeignKeys(dbSchema, fk.getTargetTable(), targetTabName, useTargetAlias);
+    if (!childList.isEmpty()) {
+      if (!list.isEmpty() && (list.size() > (childList.size() + 1))) {
+        list.clear();
+      }
+      list.add(new DbFkInfo(srcTab.getTableName(), fk.getFieldName(), fk.getTargetTable()));
+      list.addAll(childList);
+    }
+  }
+
+  /**
+   * Find a list of {@link ForeignKeys} from source table alias and target table.
+   *
+   * @param dbSchema
+   * @param srcTabAlias
+   * @param targetTabName
+   * @return
```

Comment:
Can you document what is returned if nothing found?

Reply:
done